Sources/Moya/Response.swift

```diff
@@ -122,21 +122,13 @@ public extension Response {
     /// - parameter atKeyPath: Optional key path at which to parse object.
     /// - parameter using: A `JSONDecoder` instance which is used to decode data to an object.
     func map<D: Decodable>(_ type: D.Type, atKeyPath keyPath: String? = nil, using decoder: JSONDecoder = JSONDecoder()) throws -> D {
-        let jsonData: Data
-        if let keyPath = keyPath {
-            guard let jsonDictionary = (try mapJSON() as? NSDictionary)?.value(forKeyPath: keyPath) as? [String: Any] else {
-                throw MoyaError.jsonMapping(self)
-            }
-            do {
-                jsonData = try JSONSerialization.data(withJSONObject: jsonDictionary, options: [])
-            } catch {
-                throw MoyaError.jsonMapping(self)
-            }
-        } else {
-            jsonData = data
-        }
+        let jsonData = data
```

Comment:
why this assignment?

Reply:
stayed from the original code, describes data intention. Agreed that it can be removed